artiq/firmware/libboard_misoc/config.rs

```diff
@@ -178,6 +178,14 @@ mod imp {
         })
     }
 
+    pub fn read_channel_name<F: FnOnce(Result<&str, Error>) -> R, R>(channel: &str, f: F) -> R {
+        read_str(channel, |result| f(match result {
+            Ok("") => Ok("No Name"),
+            Ok(name) => Ok(name),
+            _ => Ok("No Name")
+        }))
+    }
+
```

Comment:
Merge ``channel_number_str`` into this and move into ``rtio_mgt.rs``
Not sure about this match statement.

Reply:
And this no longer needs to be generic. It should simply return a string.